projects/batfish/test/org/batfish/main/BatfishTest.java

```diff
@@ -0,0 +1,97 @@
+package org.batfish.main;
+
+import org.batfish.common.BatfishException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.collection.IsEmptyCollection.empty;
+import static org.junit.Assert.assertThat;
+
+public class BatfishTest {
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void throwsExceptionWithSpecificType() {
+        thrown.expect(BatfishException.class);
+        thrown.expectMessage("Failed to walk path: ./test/notExist");
+        Batfish.listAllFiles(Paths.get("./test/notExist"));
+    }
+
+    @Test
+    public void testNoFileUnderPath() throws IOException {
+        File emptyFolder= folder.newFolder("emptyFolder");
+        List<Path> resultList = Batfish.listAllFiles(emptyFolder.toPath());
+        assertThat(resultList, empty());
+    }
+
+    @Test
+    public void testReadStartWithDotFile() throws IOException {
+        File startWithDot= folder.newFolder("startWithDot");
+        File file = new File(startWithDot + "/.cfg");
+        file.getParentFile().mkdir();
+        assertThat(file.createNewFile(), is(true));
+        List<Path> resultList = Batfish.listAllFiles(startWithDot.toPath());
+        assertThat(resultList, is(empty()));
+    }
+
+    @Test
+    public void testReadUnNestedPath() throws IOException {
+        File unNestedFolder= folder.newFolder("unNestedDirectory");
+        List<Path> fileList = new ArrayList<>();
```

Comment:
rename `fileList` to `expected`:

1. It's paths, not files.
2. Do not put object types in their names. aka, no `List`.